apps/lib/apps.c

```diff
@@ -2370,16 +2366,29 @@ int raw_write_stdout(const void *buf, int siz)
  * does impact behavior on some platforms, such as differentiating between
  * text and binary input/output on non-Unix platforms
  */
+static BIO *check_bio(BIO *bp, const char *what, int mode)
+{
+    if (bp == NULL) {
+        BIO_printf(bio_err, "%s: Could not open %s in mode 0x%x\n",
+                opt_getprog(), what, mode);
+        if (bio_err != NULL)
+            ERR_print_errors(bio_err);
+        exit(1);
+    }
+    return bp;
+}
+
 BIO *dup_bio_in(int format)
 {
-    return BIO_new_fp(stdin,
-                      BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+
+    return check_bio(BIO_new_fp(stdin, mode), "stdin", mode);
 }
 
 BIO *dup_bio_out(int format)
 {
-    BIO *b = BIO_new_fp(stdout,
-                        BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+    BIO *b = BIO_new_fp(stdout, mode);
```

Comment:
should we checke that b != NULL ?

Reply:
Yes, adding check.

Reply:
the check is still needed, BIO_push would happily push NULL as second parameter.
unclear what the sematic would be only the BIO_CTRL_PUSH could do something.

Reply:
fixup commit added.